Fix Consistency06 test scenario MULT-SOA-MNAMES-NO-DEL-UNDEL-2

marc-vanderwal opened the pull request:
## Purpose

This PR fixes a handful of minor errors in a test scenario for Consistency06, in the specification and the configuration files that implement it. Together, they prevented the correct operation of scenario MULT-SOA-MNAMES-NO-DEL-UNDEL-2.

## Context

Refactoring of Zonemaster-Engine unit tests.

## Changes

- Fix the test scenario specification;
- Fix the test scenario implementation (CoreDNS configuration and zone files);
- Update the sample `zonemaster-cli` output.

## How to test this PR

N/A.

marc-vanderwal committed:
Fix undelegated data in Consistency06 test scenario

In the specification for test scenarios for Consistency06, scenario
MULT-SOA-MNAMES-NO-DEL-UNDEL-2 erroneously lists
ns3.mult-soa-mnames-no-del-undel-2.consistency06.xb twice in the
undelegated data. The second occurrence should be ns4, not ns3.

Right above, the description of what SOA MNAME field values appear
mention name servers named ns1 and ns2, where ns3 and ns4 was probably
meant.

Unfortunately for us, fixing the first discrepancy does not fix the
underlying problem: it instead underscores that there is a bug in the
current implementation of Consistency06.

matsduf commented:
The specification says ns1 and ns2, seems to use ns3 and ns4. The specification needs an update.

marc-vanderwal commented:
Yes indeed, there is another problem in that scenario specification.

The first bullet point says “NS are out-of-bailiwick, "ns3.mult-soa-mnames-no-del-undel-2.consistency06.xb" and "ns4.mult-soa-mnames-no-del-undel-2.consistency06.xb"”. But the next bullet points talk about “ns1” and “ns2”.

All we really need is for ns3 and ns4 to return SOA records that differ in their MNAME fields.

marc-vanderwal committed:
Fix scenario MULT-SOA-MNAMES-NO-DEL-UNDEL-2

Looking carefully at the logs generated by CoreDNS shows that there were
some issues with the zone files and the configuration that prevented
that scenario from actually working correctly.

Fortunately, fixing these typos now fixes the zonemaster-cli run.

marc-vanderwal committed:
Fix some harmless typos

Caught a few more typos in Consistency06 test scenario specifications.

marc-vanderwal commented:
All is well. It turns out I needed to hunt down typos in more places that I thought. Now the scenario works again.

I’ll make a PR that rerecords `t/Test-consistency06.data` in Zonemaster-Engine and marks the scenario as testable.

matsduf commented:
> Shouldn’t these `template` blocks refer to ns3 and ns4 respectively, instead of ns1 and ns2?

No. "ns1" and "ns2" are IB names in the delegation no present in the undelegated zone. Note NXDOMAIN. "ns3" and "ns4" are OOB and cannot belong to the zone.
